### PR TITLE
Adds removal of orphaned entities

### PR DESCRIPTION
**Pull request by blafond**
Fixes [#702](https://github.com/hibernate/hibernate-reactive/issues/702)

### CODE REVIEW

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/LazyReplaceOrphanedEntityTest.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+		configuration.addAnnotatedClass( Campaign.class );
+		configuration.addAnnotatedClass( ExecutionDate.class );
+		configuration.addAnnotatedClass( Schedule.class );
+
+		configuration.setProperty( AvailableSettings.STATEMENT_BATCH_SIZE, "5");
+		return configuration;
+	}
+
+	@Before
+	public void prepareOrmFactory() {
```
Comment (DavideD) [inline hibernate-reactive-core/src/test/java/org/hibernate/reactive/LazyReplaceOrphanedEntityTest.java:59]: This is not necessary

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/LazyReplaceOrphanedEntityTest.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import io.vertx.ext.unit.TestContext;
+import org.assertj.core.api.Assertions;
+
+public class LazyReplaceOrphanedEntityTest extends BaseReactiveTest {
+
+	private Campaign theCampaign;
+
+	private SessionFactory ormFactory;
```
Comment (DavideD) [inline hibernate-reactive-core/src/test/java/org/hibernate/reactive/LazyReplaceOrphanedEntityTest.java:45]: You don't need the ORM factory

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/LazyReplaceOrphanedEntityTest.java [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+	}
+
+	@Entity (name="Campaign")
+	public static class Campaign implements Serializable {
+
+		@Id @GeneratedValue
+		private Integer id;
+
+		@OneToOne(mappedBy = "campaign",  cascade = CascadeType.ALL, fetch = FetchType.LAZY, orphanRemoval = true)
+		@JsonIgnore
```
Comment (DavideD) [inline hibernate-reactive-core/src/test/java/org/hibernate/reactive/LazyReplaceOrphanedEntityTest.java:138]: These `JsonIgnore` annotations are not necessary

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/LazyReplaceOrphanedEntityTest.java [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+		@Id
+		@Column(name = "id")
+		private String id = UUID.randomUUID().toString();
+
+		@Column(name = "code_name")
+		private String code_name;
+
+		@OneToOne
+		@JoinColumn(name = "campaign_id")
+		@JsonIgnore
```
Comment (DavideD) [inline hibernate-reactive-core/src/test/java/org/hibernate/reactive/LazyReplaceOrphanedEntityTest.java:174]: No need for the `@JsonIgnore`

**Review by DavideD — CHANGES_REQUESTED** [review]

**Review by gavinking — APPROVED** [review]
This LGTM.

**Pull request merged**